src/Components/AddTask/AddTask.tsx

```diff
@@ -1,68 +1,136 @@
 import axios from 'axios'
-import { useContext } from 'react'
+import { useContext, useEffect, useState } from 'react'
 import { useForm } from 'react-hook-form'
 import { useNavigate } from 'react-router-dom'
 import { AuthContext } from '../../Context/AuthContext'
+import { ToastContext } from '../../Context/ToastContext'
+import { TasksContext } from '../../Context/TasksContext'
 
-export default function AddProject() {
- const navigate= useNavigate()
-const {register,handleSubmit,formState:{errors}}=useForm()
-let{baseUrl,reqHeaders}:any=useContext(AuthContext)
-const AddProject =(data)=>{
-axios.post(`${baseUrl}/Task`,data,{headers:reqHeaders}).then((response)=>{
-  
-  navigate(-1)
-  
-}).catch((error)=>{
-  console.log(error);
+export default function AddTask() {
+  const navigate = useNavigate()
+  const { register, handleSubmit, formState: { errors } } = useForm()
+  let { baseUrl, reqHeaders }: any = useContext(AuthContext)
+  let { getToastValue } = useContext(ToastContext)
+  const [users, setUsers] = useState([])
+  const [allProjects, setAllProjects] = useState([])
+  // let{  getAllTasks,getDataTasks}=useContext(TasksContext)
+ 
+
+
+
+  {/* Add  Task*/ }
+  const AddTask = (data) => {
+    axios.post(`${baseUrl}/Task`, data, { headers: reqHeaders }).then((response) => {
+
+      getToastValue('success','Added Successfully')
+      navigate('/dashboard/tasks')
   
-})
 
-}
+      
+
+    
+      
+
+    }).catch((error) => {
+
+      getToastValue("error", error?.response?.data?.message)
+
+    })
+
+  }
+
+  {/* Get  Users*/ }
+  const Users = () => {
+    axios.get(`${baseUrl}/Users/Manager?pageSize=10&pageNumber=1`, { headers: reqHeaders }).then((response) => {
+      setUsers(response?.data?.data)
+
+    }).catch((error) => {
+      getToastValue('error', error?.response?.data?.message)
+
+    })
+  }
+
+  {/* Get  All Projects*/ }
+  const AllProjects = () => {
```

Comment:
Consider changing the number of the function from a noun to a verb to indicate that it is a function:

instead of `AllProjects` => `getAllProjects`